### ISSUE TITLE
Auto discovery support

### ISSUE THREAD

**Issue by Kazhuu** [piotrbulinski/flexit_bacnet#13]
Could this feature to be possible to add? I'd be willing to help on this one if needed. I opened this issue here so we can collaborate publicly and share information which others can also see.

**Comment by lellky** [comment]
A wild thought: is it possible to scan the network and see if devices with a matching pattern for Flexit mac-addresses is found? Then maybe this library does not have to support it at all, but the flexit integration in Home Assistant handles it?

**Comment by piotrbulinski** [comment]
@Kazhuu, I've explored auto-discovery in BACnet, but I didn't immediately find a solution for it in the specifications. My conclusion is that I would need more time to figure it out, and unfortunately, I don't have that time available right now.

If you'd like to give it a try, please feel free to do so. I'll be happy to review your work whenever you're ready.

Please keep in mind that I would prefer not to rely on external packages. If using an external package becomes necessary, I'd prefer to document how to perform auto-discovery with it and leave it up to the user of this package to decide whether they want to install any external dependencies or not.

**Comment by lellky** [comment]
I might have been a bit too optimistic. Discovering bacnet devices seems to be done on the BACNet protocol (obviously). So to be able to discover Flexit devices on the network we cannot do it from home assistant without using a bacnet library e.g. bacpypes or flexit_bacnet. I think it would make sense to integrate it into flexit_bacnet actually.

This is a screenshot from Wireshark when I open the flexit app in my android phone (192.168.2.144) and it discovers my Flexit Nordic S4 (192.168.2.14):
<img width="1543" alt="Screenshot 2023-12-02 at 09 42 48" src="https://github.com/piotrbulinski/flexit_bacnet/assets/7119913/4efa680f-017c-43e8-b0c1-feab5a4ee740">

**Comment by piotrbulinski** [comment]
@lellky it looks like it's sending a broadcast message (see IP destination) - I've developed a fair share of python code in my life, but that was a long time ago and never anything with low level networking, so my first litmus-test for the feasibility of this approach would be to check if we can even send broadcast UDP packets using Python without any issues (e.g. do we need root access to send to broadcast address or can any user do it?). If you can figure this out, I can then help implementing the required payload of the packet and parsing the response.

**Comment by lellky** [comment]
@piotrbulinski I dont understand one percent of network/bacnet/broadcast. But I can google.

This code (found on stack overflow and modified):
```
import socket

PORT = 47808
BROADCAST_IP = "192.168.2.255"
msg = b"Hello, World!"

def main():
    with socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP) as sock:
        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
        sock.sendto(msg, (BROADCAST_IP, PORT))

if __name__ == "__main__":
    main()
```

seems to send out a broadcast message on the network:
<img width="1336" alt="Screenshot 2023-12-03 at 20 58 55" src="https://github.com/piotrbulinski/flexit_bacnet/assets/7119913/7cbb0fd9-b06c-4b4f-945d-5d6999ff2dd4">

I have also looked at the source code for BAC0. There are functions in the code for who-is/i-am. Maybe we can find inspiration from there? https://github.com/JoelBender/bacpypes/blob/a5be2ad5ac69821c12299716b167dd52041b5342/py34/bacpypes/service/device.py#L22

**Comment by piotrbulinski** [comment]
@lellky I can look into this during the holidays.

**Comment by lellky** [comment]
@piotrbulinski please let me know if I can help in any way!

**Comment by piotrbulinski** [comment]
@lellky @Kazhuu I've made some progress on that in #15 - please take a look, test and let me know if that would work for you.

It will currently only work on Linux though.

**Comment by Kazhuu** [comment]
> @lellky @Kazhuu I've made some progress on that in #15 - please take a look, test and let me know if that would work for you.
> 
> It will currently only work on Linux though.

Hello! I'll test this on first week of January when I get home. Great progress!

**Comment by piotrbulinski** [comment]
@Kazhuu @lellky I've released the code as 2.2.0, so you can try to use that in Home Assistant. If you identify any issues with the mechanism, please open a new issue.

Happy Holidays!

**Comment by lellky** [comment]
Great work!

I will play around with the feature in Home Assistant and see if I can get something working.

Happy holidays and great work!
